ExtractLightsMerge

```diff
@@ -0,0 +1,392 @@
+/* -*-c++-*- */
+#pragma once
+
+#ifndef EXTRACT_LIGHTS_MERGE_H
+#define EXTRACT_LIGHTS_MERGE_H 1
+
+
+/**
+ * convert Env map Regions to Lights
+ */ 
+void createLightsFromRegions(const std::vector<satRegion>& regions, std::vector<light>& lights, const float *rgba, const uint width, const uint height, const uint nc)
```

Comment:
use typedef instead of inline template so something like 
void createLightsFromRegions(const std::vector<satRegion>& regions, std::vector<light>& lights, const float *rgba, const uint width, const uint height, const uint nc)

should be 
typedef std::vector<satRegion> SatRegionVector;
typedef std::vector<light> LightVector;

void createLightsFromRegions(const SatRegionVector& regions, LightVector& lights, const float *rgba, const uint width, const uint height, const uint nc)

it follows the same coding style than osg core files